### PR TITLE
i.ann.maskrcnn: remove deprecated distutils

### PR DESCRIPTION
**Pull request by ninsbl**
See also: https://github.com/OSGeo/grass/issues/2601

### CODE REVIEW

File: src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py [hunk 1]
```diff
@@ -29,10 +29,15 @@
 import utils
 
 # Requires TensorFlow 1.3+ and Keras 2.0.8+.
-from distutils.version import LooseVersion
-
-assert LooseVersion(tf.__version__) >= LooseVersion("1.3")
-assert LooseVersion(keras.__version__) >= LooseVersion("2.0.8")
+version_pattern = "([0-9]{1,3}).([0-9]{1,4})"
```
Comment (pesekon2) [inline src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py:32]:
```suggestion
version_pattern = "([0-9]{1}).([0-9]{1,2})"
```

File: src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py [hunk 2]
```diff
@@ -29,10 +29,15 @@
...
+assert list(map(int, re.match(version_pattern, tf.__version__).groups())) >= [1, 3]
+version_pattern = "([0-9]{1,3}).([0-9]{1,4}).([0-9]{1,4})"
```
Comment (pesekon2) [inline src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py:34]:
```suggestion
version_pattern = "([0-9]{1}).([0-9]{1,2}).([0-9]{1,2})"
```

File: src/imagery/i.ann.maskrcnn/maskrcnnlib/utils.py [hunk 3]
```diff
@@ -979,7 +978,9 @@ def resize(
     of skimage. This solves the problem by using different parameters per
     version. And it provides a central place to control resizing defaults.
     """
-    if LooseVersion(skimage.__version__) >= LooseVersion("0.14"):
+    if list(
+        map(int, re.match("([0-9]{1,3}).([0-9]{1,4})", skimage.__version__).groups())
```
Comment (pesekon2) [inline src/imagery/i.ann.maskrcnn/maskrcnnlib/utils.py:982]:
```suggestion
        map(int, re.match("([0-9]{1}).([0-9]{1,2})", skimage.__version__).groups())
```

File: src/imagery/i.ann.maskrcnn/maskrcnnlib/utils.py [hunk 4]
```diff
@@ -24,7 +24,6 @@
 import urllib.request
 import glob
 import warnings
```
Comment (pesekon2) [inline src/imagery/i.ann.maskrcnn/maskrcnnlib/utils.py:26]:
```suggestion
import re
import warnings
```

**Review by pesekon2 — CHANGES_REQUESTED** [review]
* add missing `import re`
* remove unnecessarily big sizes of regex strings (first number just one int, then two ints)

File: src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py (hunk 1 above)
Comment (ninsbl) [reply src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py:32]: I agree that the likelyhood that broad version ranges are needed is low. However, it should not hurt either and with wider ranges you are less vulnerable. But I have nothing against reducing the range if you prefer that...

Comment (pesekon2) [reply src/imagery/i.ann.maskrcnn/maskrcnnlib/model.py:32]: Okay. In the end, I have no strong opinion on that. So let's keep the broader ranges. Approving the PR.

**Review by pesekon2 — APPROVED** [review]

**Comment by ninsbl** [comment]
Thanks for reviewing, @pesekon2

**Pull request merged**